### DISCUSSION TITLE
icone no appbar leading e no action com o mesmo tamanho

### DISCUSSION THREAD

**Discussion by Dablinlabs** [Flutterando/forum#24] [Perguntas e Respostas] [answered]
salve galera bom dia tudo firmeza ?
galera sou iniciante e estou com um pequeno problema na minha appbar 
https://ibb.co/QrnskDp
não estou conseguindo fazer o leading ficar do mesmo tamanho que no meu action, mesmo utilizando o mesmo componente para criar

```
//meu appBar
appBar: AppBar(
          backgroundColor: Colors.transparent,
          elevation: 0,
          toolbarHeight: 70,
          leading: AppBarCircleItem(
            icon: Icons.arrow_back_ios_new,
            press: () {
              Navigator.of(context).pop();
            },
          ),
          actions: [
            AppBarCircleItem(
              icon: Icons.shopping_cart,
              press: () {},
            ),
          ],
        ),

//codigo usado para criar o botao
  final IconData? icon;
  final GestureTapCancelCallback press;

  const AppBarCircleItem({super.key, this.icon, required this.press});

  @override
  Widget build(BuildContext context) {
    return Container(
      width: 45,
      height: 45,
      margin: EdgeInsets.symmetric(
          horizontal: 10,
          vertical: 10),
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(40),
        color: Theme.of(context).colorScheme.primary,
      ),
      child: Padding(
        padding: const EdgeInsets.all(3),
        child: IconButton(
          onPressed: press,
          icon: Icon(
            icon,
            color: Theme.of(context).colorScheme.secondary,
          ),
        ),
      ),
    );
  }
}

```

**Comment by mateusfccp** [comment]
O código provido não contém nenhum `AppBar`, tem certeza que o código que você colou é o correto?

**Reply by Dablinlabs** [reply]
Esse código é o componente que estou criando o botão
O que tem o app bar tem so o 
Leading: criarbotao(icone, () {}),
Actions: [
     criarbotao(icone, () {}),
]

**Reply by mateusfccp** [reply]
Só com essas informações fica difícil de ajudar. O problema pode estar sendo causado, por exemplo, pelo `getProportionateScreenWidth`, que não sabemos do que se trata. Fora isso, não vejo nada errado, mas não posso ter certeza só com essa pouca informação.

Tenta fazer um exemplo mínimo reproduzível para que possamos te ajudar melhor.

**Reply by Dablinlabs** [reply]
perdão, coloquei o codigo com a appbar

**Comment by eduardoflorence** [comment, marked as answer]
@Lanysc a AppBar possui a propriedade `leadingWidth`, que possui valor padrão igual a 56. Altere o valor dela para se adaptar a largura que você deseja para o seu leading.

**Comment by cbalmeida** [comment]
Acredito que seja melhor deixar o appbar gerenciar os tamanhos 

Já tentou tirar os tamanhos que voce definiu no container e ver o que acontece?

    //  width: 45,
    //  height: 45,
